doc/snmp-schema-addition.md

```diff
@@ -0,0 +1,125 @@
+
+## Proposal: integrate the SNMP configuration into config DB ##
+### Proposed Schema ###
+
+```
+{
+   "SNMP":{
+      "LOCATION":{
+         "LOCATION":"<SNMP_LOCATION_STRING>"
+      },
+      "CONTACT":{
+         "CONTACT":"<SNMP_CONTACT_STRING>"
+      }
+   }
+}
+{
+   "SNMP_COMMUNITY":{
+      "<SNMP_COMM>":{
+         "TYPE":"RO|RW"
+      },
+      "<SNMP_COMM>":{
+         "TYPE":"RO|RW"
+      },
+      "<SNMP_COMM>":{
+         "TYPE":"RO|RW"
+      }
+   }
+}
+{
+   "SNMP_USER":{
+      "<SNMP_USER>":{
+         "SNMP_USER_STRING":"<SNMP_USER_STRING>",
+         "SNMP_USER_TYPE":"noAuthNoPriv|AuthNoPriv|Priv",
+         "SNMP_USER_AUTH_TYPE":"MD5|SHA|HMAC-SHA-2",
+         "SNMP_USER_ENCRYPTION_TYPE":"DES|AES",
+         "SNMP_USER_AUTH_PASSWORD":"<AUTH_PASSWORD_STRING>",
+         "SNMP_USER_ENCRYPTION_PASSWORD":"<ENCRYPTION_PASSWORD_STRING>"
+      },
+      "<SNMP_USER>":{
+         "SNMP_USER_STRING":"<SNMP_USER_STRING>",
+         "SNMP_USER_TYPE":"noAuthNoPriv|AuthNoPriv|Priv",
+         "SNMP_USER_AUTH_TYPE":"MD5|SHA|HMAC-SHA-2",
+         "SNMP_USER_ENCRYPTION_TYPE":"DES|AES",
+         "SNMP_USER_AUTH_PASSWORD":"<AUTH_PASSWORD_STRING>",
+         "SNMP_USER_ENCRYPTION_PASSWORD":"<ENCRYPTION_PASSWORD_STRING>"
+      },
+      "<SNMP_USER>":{
+         "SNMP_USER_STRING":"<SNMP_USER_STRING>",
+         "SNMP_USER_TYPE":"noAuthNoPriv|AuthNoPriv|Priv",
+         "SNMP_USER_AUTH_TYPE":"MD5|SHA|HMAC-SHA-2",
+         "SNMP_USER_ENCRYPTION_TYPE":"DES|AES",
+         "SNMP_USER_AUTH_PASSWORD":"<AUTH_PASSWORD_STRING>",
+         "SNMP_USER_ENCRYPTION_PASSWORD":"<ENCRYPTION_PASSWORD_STRING>"
+      }
+   }
+}
+```
+
+Where:
+- LOCATION_STRING:  String, defines the snmpLocation, default: "".  
+- CONTACT_STRING:   String, defines the snmpContact, default: "" or the current hardcoded value for backwards compatibility.  
+- COMMUNITY_STRING: String, defines the community string.
+- TYPE: String, defines the community string permissions of either read-only RO or read-write RW.
+- SNMP_USER: String, defines the SNMP user.
+- SNMP_USER_TYPE: String, defines which authentication and encryption methods will be used for that SNMP user.  The options are noAuthNoPriv or AuthNoPriv or Priv.
+- SNMP_USER_AUTH_TYPE: String, defines which authentication type will be used for that SNMP user MD5 or SHA or MHAC-SHA-2.
+- SNMP_USER_ENCRYPTION_TYPE: String, defines which encryption type will be used for that SNMP user DES or AES. 
+- SNMP_USER_AUTH_PASSWORD: String, defines which authentication password will be used for that SNMP user.
+- SNMP_USER_ENCRYPTION_PASSWORD: String, defines which encryption password will be used for that SNMP user. 
+
+
+
+New keys:
+- "TYPE":  Optional, if ommited defaults to 'ro', there are 2 possible values:  
+           "RO": read-only, the only implemented method at this time.  
+           "RW": well you never know - here for completeness but unused in the code.  
```

Comment:
These are not indented in the rendered Markdown. Suggest making them bullet points.